lib/fetch/index.js

```diff
@@ -1554,9 +1563,85 @@ function httpNetworkFetch (
 
           context.on('terminated', onResponseAborted)
 
+          const codings = headers.get('content-encoding')
+          if (codings === 'gzip' || codings === 'x-gzip') {
+            this.decoder = zlib.createGunzip()
+          } else if (codings === 'deflate' || codings === 'x-deflate') {
+            this.decoder = zlib.createInflate()
+          } else if (codings === 'br') {
+            this.decoder = zlib.createBrotliDecompress()
+          } else {
+            // TODO: What to do when coding is unsupported?
```

Comment:
@mcollina @szmarczak What to do when coding is unsupported?

Reply:
Got returns raw buffer. I guess `fetch` would do this as well. Let me check

Reply:
Also it's a little weird that the end user has no way of determining whether the response has been decoded or not.

Reply:
@szmarczak would you mind checking what chrome does?

Reply:
Returns the value with no errors.

Reply:
> Also it's a little weird that the end user has no way of determining whether the response has been decoded or not.

I agree. Can you ask someone from the `fetch` team about this?